internal/forge/gitlab/client.go

```diff
@@ -10,8 +14,18 @@ type gitlabClient struct {
 	Users            usersService
 }
 
-func newGitLabClient(baseURL string, accessToken string) (*gitlabClient, error) {
-	client, err := gitlab.NewClient(accessToken, gitlab.WithBaseURL(baseURL))
+func newGitLabClient(baseURL string, tok *AuthenticationToken) (*gitlabClient, error) {
+	var newClient func(string, ...gitlab.ClientOptionFunc) (*gitlab.Client, error)
+	switch tok.AuthType {
+	case AuthTypePAT:
+		newClient = gitlab.NewClient
+	case AuthTypeOAuth2:
+		newClient = gitlab.NewOAuthClient
```

Comment:
this is the important bit: gitlab library uses a different client based on the token type, so we need to record what auth method was used: pat or oauth.